components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2838,6 +2851,15 @@ private FederatedAuthenticatorConfig createFederatedAuthenticatorConfig(String f
         authConfig.setName(authenticatorName);
         authConfig.setDisplayName(getDisplayNameOfAuthenticator(authenticatorName));
         authConfig.setEnabled(authenticator.getIsEnabled());
+        // Resolve definedBy type: If there is authenticator by same name and its type is system: SYSTEM. If not: USER.
+        FederatedAuthenticatorConfig authenticatorConfig = ApplicationAuthenticatorService.getInstance()
+                .getFederatedAuthenticatorByName(authenticatorName);
+        if (authenticatorConfig != null &&
+                DefinedByType.SYSTEM.equals(authenticatorConfig.getDefinedByType())) {
+            authConfig.setDefinedByType(DefinedByType.SYSTEM);
+        } else {
+            authConfig.setDefinedByType(DefinedByType.USER);
+        }
```

Comment:
extract to a different method

Reply:
Addressed with afdfbaea310d63706424d6594ad2cae5fe86b80a